core/test/base/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+set (link_libs Core)
+
+ROOT_ADD_GTEST(CoreBaseTests
```

Comment:
I'd prefer to have src/, inc/, test/ at the same level. E.g. sql/mysql would only be enabled if we have the mysql library in the system; that would automatically enable / disable the tests with it.